sdcm/utils/version_utils.py

```diff
@@ -2,66 +2,45 @@
 
 import requests
 from pkg_resources import parse_version
-from repodataParser.RepoParser import Parser
-
 
 # Examples of ScyllaDB version strings:
 #   - 666.development-0.20200205.2816404f575
 #   - 3.3.rc1-0.20200209.0d0c1d43188
 #   - 2019.1.4-0.20191217.b59e92dbd
+from sdcm.utils.url_validate import verify_url_is_valid
+from sdcm.utils.xml_utils import get_all_scylla_versions
+
 SCYLLA_VERSION_RE = re.compile(r"\d+(\.\d+)?\.[\d\w]+([.~][\d\w]+)?")
 
 
 def get_branch_version(url):
-    try:
-        return get_branch_version_from_list(url)
-    except ValueError:
-        pass
-
-    try:
-        return get_branch_version_from_repo(url)
-    except ValueError:
-        raise ValueError("url isn't a correct deb list or yum repo\n\turl:[{}]".format(url))
+    get_version_method = get_branch_version_from_list
+    if url.endswith(('.list', '.repo')):
```

Comment:
why adding this limit of file extension ?

Reply:
Because the first URL must contain one of the following files.
Also, we have different logic for each file.

Reply:
i don't see any logic for .list files